substrate/primitives/staking/src/lib.rs

```diff
@@ -463,6 +500,39 @@ pub struct PagedExposureMetadata<Balance: HasCompact + codec::MaxEncodedLen> {
 	pub page_count: Page,
 }
 
+impl<Balance> PagedExposureMetadata<Balance>
+where
+	Balance: HasCompact
+		+ codec::MaxEncodedLen
+		+ Add<Output = Balance>
+		+ Sub<Output = Balance>
+		+ sp_runtime::Saturating
+		+ PartialEq
+		+ Copy
+		+ sp_runtime::traits::Debug,
+{
+	/// Consomes self and returns the result of the metadata updated with `other_balances` and
+	/// of adding `other_num` nominators to the metadata.
+	///
+	/// `Max` is a getter of the maximum number of nominators per page.
+	pub fn update_with<Max: sp_core::Get<u32>>(
+		self,
+		others_balance: Balance,
+		others_num: u32,
+	) -> Self {
+		let new_nominator_count = self.nominator_count.saturating_add(others_num);
+		let new_page_count =
+			new_nominator_count.saturating_add(Max::get()).saturating_div(Max::get());
```

Comment:
The formula should be this
```suggestion
			new_nominator_count
			    .defensive_saturating_add(Max::get())
			    .defensive_saturating_sub(1)
			    .saturating_div(Max::get());
```